### PR TITLE
update Mixin.js to support adding steps asynchronously

### PR DESCRIPTION
**Pull request by gilbarbara**

### CODE REVIEW

File: lib/js/Mixin.js [hunk 1]
```diff
@@ -74,12 +76,18 @@ module.exports = function(settings, done) {
...
+      }
       cb = cb || function() {};
+      this.settings.steps = steps;
 
-      this.setState(this.getInitialState(), function() {
-        this.settings.steps = steps;
+      this.setState({
+        currentIndex: this.state.currentIndex < 0 ? 0 : this.state.currentIndex,
+        setTourSteps: steps.length
+      }, function() {
```
Comment (jakemmarsh) [inline lib/js/Mixin.js]: cb can be passed directly here, rather than wrapped inside the `function()`

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# EditorConfig helps developers define and maintain consistent
```
Comment (jakemmarsh) [inline .editorconfig]: can you remove this file?

**Comment by jakemmarsh** [comment]
Looks good, will merge pending a few notes!

**Comment by gilbarbara** [comment]
Hey,
I pushed the requested changes to this PR. :+1: 

But I made a joyride mixin myself since I would need a few other features.. In case you want to check it: https://github.com/gilbarbara/react-joyride/
Credits for you in the bottom.

cheers

**Pull request merged**